website_form_recaptcha/models/website_config_settings.py

```diff
@@ -1,15 +1,34 @@
 # Copyright 2019 Simone Orsi - Camptocamp SA
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 
-from odoo import fields, models
+from odoo import api, fields, models
 
 
 class ResConfigSettings(models.TransientModel):
     _inherit = 'res.config.settings'
 
     recaptcha_key_site = fields.Char(
         related='website_id.recaptcha_key_site',
+        readonly=False,
     )
     recaptcha_key_secret = fields.Char(
         related='website_id.recaptcha_key_secret',
+        readonly=False,
+    )
+
+
+    @api.depends('website_id')
+    def has_google_recaptcha(self):
+        self.has_google_recaptcha = bool(self.recaptcha_key_site)
+
+    def inverse_has_google_recaptcha(self):
+        if not self.has_google_recaptcha:
+            self.recaptcha_key_site = False
+            self.recaptcha_key_secret = False
+
+    has_google_recaptcha = fields.Boolean(
+        "Google reCaptcha",
+        compute=has_google_recaptcha,
+        inverse=inverse_has_google_recaptcha,
```

Comment:
Field definitions should go at the beginning of the class declaration per guideline.

Use a string that indicates the method names for compute and inverse instead of the raw method object.